universe/Building.cpp

```diff
@@ -239,13 +239,20 @@ float BuildingType::ProductionCost(int empire_id, int location_id) const {
         if (m_production_cost && m_production_cost->ConstantExpr())
             return m_production_cost->Eval();
 
+        const auto arbitrary_large_number = 999999.9f;
+
         std::shared_ptr<UniverseObject> location = GetUniverseObject(location_id);
-        if (!location)
-            return 999999.9f;    // arbitrary large number
+        if (!location) {
+            ErrorLogger() << "Location is missing using production cost of " << arbitrary_large_number;
```

Comment:
not necessarily an error... pedia can show stuff even when there's no game ongoing.